copis/coms/edsdk_object.py

```diff
@@ -13,265 +13,317 @@
 # You should have received a copy of the GNU General Public License
 # along with COPISClient.  If not, see <https://www.gnu.org/licenses/>.
 
-import logging
-import os
-import time
-from ctypes import *
-
-from canon.EDSDKLib import *
-
-_edsdk = None
-_console = None
-_camlist = None
-_camref = None
-_camindex = -1
-running = False
-
-num_cams = 0
-
-waiting_for_image = False
-image_filename = None
-image_folder = ''
-image_prefix = 'COPIS'
-image_index = None
+"""Manages COPIS EDSDK Communications"""
```

Comment:
Will test out your branch to see if Edsdk works properly, I'll resolve this comment once I do that

Reply:
Update: tested, and you're right that it doesn't work. I think the reason is because you moved `_handle_object`, `_handle_property`, and `_handle_state` into `LocalEDSDK`, turning them into class methods. I don't think the EDSDK likes that, and they only work when defined as standalone functions - which then makes wiring up error messages and downloading the image messy. Maybe we should rethink `edsdk_object` a little?

Reply:
Just tested - I confirm that it works!